cmd/clusterctl/pkg/client/common.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client/repository"
+)
+
+// getComponentsByName is a utility methods that returns provider component for a given provider, targetNamespace and WatchingNamespace.
+func (c *clusterctlClient) getComponentsByName(provider string, targetNamespace string, watchingNamespace string) (repository.Components, error) {
+
+	// parse the abbreviated syntax for name[:version]
+	name, version, err := parseProviderName(provider)
+	if err != nil {
+		return nil, err
+	}
+
+	// if a target namespace is defined, use it, otherwise let the namespace empty so the default one defined in the
+	// provider components YAML will be used
+	var namespace string
```

Comment:
I'm not sure you need this logic. I think you can just pass `targetNamespace` in on line 58 - if it's the empty string, it will use the default from the yaml; if it's not the empty string, it will use the value from the variable. Make sense?

Reply:
yes! done